app/code/Magento/Sales/Api/Data/OrderPaymentInterface.php

```diff
@@ -261,7 +261,7 @@ public function getAdditionalData();
     /**
      * Gets the additional information for the order payment.
      *
-     * @return string[] Array of additional information.
+     * @return array|null|mixed Array of additional information.
```

Comment:
Changing annotation doesn't fix the problem in global scope. It fixes an issue for rest API but brokes payment methods output. The common issue more deeply than that annotation and related to how some payment methods use additional information.

It's known issue and requires deep refactoring for some payment methods, like PayPal and Authorize.net. After that refactoring this annotation also will be fixed.